pint/quantity.py

```diff
@@ -1442,48 +1437,32 @@ def __ito_if_needed(self, to_units):
 
         self.ito(to_units)
 
-    def __numpy_method_wrap(self, func, *args, **kwargs):
-        """Convenience method to wrap on the fly numpy method taking
-        care of the units.
-        """
-        if func.__name__ in self.__require_units:
-            self.__ito_if_needed(self.__require_units[func.__name__])
-
-        value = func(*args, **kwargs)
-
-        if func.__name__ in self.__copy_units:
-            return self.__class__(value, self._units)
-
-        if func.__name__ in self.__prod_units:
-            tmp = self.__prod_units[func.__name__]
-            if tmp == 'size':
-                return self.__class__(value, self._units ** self._magnitude.size)
-            return self.__class__(value, self._units ** tmp)
-
-        return value
-
     def __len__(self):
         return len(self._magnitude)
 
     def __getattr__(self, item):
         # Attributes starting with `__array_` are common attributes of NumPy ndarray.
         # They are requested by numpy functions.
         if item.startswith('__array_'):
-            warnings.warn("The unit of the quantity is stripped.", UnitStrippedWarning, stacklevel=2)
+            warnings.warn("The unit of the quantity is stripped when getting {} "
+                          "attribute".format(item), UnitStrippedWarning, stacklevel=2)
             if isinstance(self._magnitude, ndarray):
                 return getattr(self._magnitude, item)
             else:
                 # If an `__array_` attributes is requested but the magnitude is not an ndarray,
                 # we convert the magnitude to a numpy ndarray.
-                self._magnitude = _to_magnitude(self._magnitude, force_ndarray=True)
-                return getattr(self._magnitude, item)
-        elif item in self.__handled:
-            if not isinstance(self._magnitude, ndarray):
-                self._magnitude = _to_magnitude(self._magnitude, True)
-            attr = getattr(self._magnitude, item)
+                # TODO (#905 follow-up): Potentially problematic, investigate for duck arrays
+                magnitude_as_array = _to_magnitude(self._magnitude, force_ndarray=True)
+                return getattr(magnitude_as_array, item)
+        elif item in HANDLED_UFUNCS or self._wrapped_numpy_methods:
```

Comment:
I believe this should be
```python
        elif item in HANDLED_UFUNCS or item in self._wrapped_numpy_methods:
```
which would fix my problem with `ndim`

Reply:
Yep, that is what I had intended. I had a commit incoming, but I thought I'd try bundling it with any other fixes after testing it with your test branch at https://github.com/pydata/xarray/pull/3600. However, I'm still a bit overwhelmed with trying to figure out where all the errors and `__array_interface__`/`__array_struct__` calls are coming from.